core/src/main/java/io/grpc/transport/inprocess/InProcessTransport.java

```diff
@@ -136,19 +137,19 @@ public synchronized void shutdown() {
     if (shutdown) {
       return;
     }
-    shutdownStatus = Status.UNAVAILABLE.withDescription("transport was requested to shut down");
-    notifyShutdown();
+    shutdownStatus = Status.OK.withDescription("transport was requested to shut down");
```

Comment:
No, the previous status of UNAVAILABLE was correct for `shutdownStatus.` If you look at how `shutdownStatus` is used, it is the status for any streams created after the transport is shutdown. This is done instead of throwing IllegalStateException so that we can cleanly inform the user why the transport is shutdown.